diff -ruN build/core/main.mk build/core/main.mk
--- build/core/main.mk	2014-01-31 23:24:49.807633407 +0000
+++ build/core/main.mk	2014-01-31 23:24:13.339634563 +0000
@@ -631,6 +631,8 @@
   # Filter out the overridden packages before doing expansion
   product_MODULES := $(filter-out $(foreach p, $(product_MODULES), \
       $(PACKAGES.$(p).OVERRIDES)), $(product_MODULES))
+  product_MODULES := $(filter-out $(foreach p, $(product_MODULES), \
+      $(PRODUCT_PACKAGES_FILTER)), $(product_MODULES))
   $(call expand-required-modules,product_MODULES,$(product_MODULES))
   product_FILES := $(call module-installed-files, $(product_MODULES))
   ifeq (0,1)
