diff -ruN hardware/libhardware/Android.mk hardware/libhardware/Android.mk
--- hardware/libhardware/Android.mk	2014-03-01 07:48:44.283130412 +0000
+++ hardware/libhardware/Android.mk	2014-03-01 07:46:32.667755010 +0000
@@ -5,6 +5,7 @@
 
 LOCAL_PATH:= $(call my-dir)
 include $(CLEAR_VARS)
+LOCAL_CLANG := true
 
 LOCAL_SHARED_LIBRARIES := libcutils liblog
 
diff -ruN hardware/libhardware/modules/audio/Android.mk hardware/libhardware/modules/audio/Android.mk
--- hardware/libhardware/modules/audio/Android.mk	2014-03-01 07:48:44.291130373 +0000
+++ hardware/libhardware/modules/audio/Android.mk	2014-03-01 07:46:32.711754802 +0000
@@ -21,7 +21,7 @@
 # The format of the name is audio.<type>.<hardware/etc>.so where the only
 # required type is 'primary'. Other possibilites are 'a2dp', 'usb', etc.
 include $(CLEAR_VARS)
-
+LOCAL_CLANG := true
 LOCAL_MODULE := audio.primary.default
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 LOCAL_SRC_FILES := audio_hw.c
@@ -33,7 +33,7 @@
 # The stub audio policy HAL module that can be used as a skeleton for
 # new implementations.
 include $(CLEAR_VARS)
-
+LOCAL_CLANG := true
 LOCAL_MODULE := audio_policy.stub
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 LOCAL_SRC_FILES := audio_policy.c
diff -ruN hardware/libhardware/modules/audio_remote_submix/Android.mk hardware/libhardware/modules/audio_remote_submix/Android.mk
--- hardware/libhardware/modules/audio_remote_submix/Android.mk	2014-03-01 07:48:44.291130373 +0000
+++ hardware/libhardware/modules/audio_remote_submix/Android.mk	2014-03-01 07:46:32.711754802 +0000
@@ -15,7 +15,7 @@
 LOCAL_PATH := $(call my-dir)
 
 include $(CLEAR_VARS)
-
+LOCAL_CLANG := true
 LOCAL_MODULE := audio.r_submix.default
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 LOCAL_SRC_FILES := \
diff -ruN hardware/libhardware/modules/camera/Android.mk hardware/libhardware/modules/camera/Android.mk
--- hardware/libhardware/modules/camera/Android.mk	2014-03-01 07:48:44.295130354 +0000
+++ hardware/libhardware/modules/camera/Android.mk	2014-03-01 07:46:32.711754802 +0000
@@ -15,7 +15,7 @@
 LOCAL_PATH := $(call my-dir)
 
 include $(CLEAR_VARS)
-
+LOCAL_CLANG := true
 LOCAL_MODULE := camera.default
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 
diff -ruN hardware/libhardware/modules/consumerir/Android.mk hardware/libhardware/modules/consumerir/Android.mk
--- hardware/libhardware/modules/consumerir/Android.mk	2014-03-01 07:48:44.295130354 +0000
+++ hardware/libhardware/modules/consumerir/Android.mk	2014-03-01 07:46:32.711754802 +0000
@@ -15,7 +15,7 @@
 LOCAL_PATH := $(call my-dir)
 
 include $(CLEAR_VARS)
-
+LOCAL_CLANG := true
 LOCAL_MODULE := consumerir.default
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 LOCAL_SRC_FILES := consumerir.c
diff -ruN hardware/libhardware/modules/gralloc/Android.mk hardware/libhardware/modules/gralloc/Android.mk
--- hardware/libhardware/modules/gralloc/Android.mk	2014-03-01 07:48:44.295130354 +0000
+++ hardware/libhardware/modules/gralloc/Android.mk	2014-03-01 07:46:32.711754802 +0000
@@ -18,7 +18,7 @@
 # HAL module implemenation stored in
 # hw/<OVERLAY_HARDWARE_MODULE_ID>.<ro.product.board>.so
 include $(CLEAR_VARS)
-
+LOCAL_CLANG := true
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 LOCAL_SHARED_LIBRARIES := liblog libcutils
 
diff -ruN hardware/libhardware/modules/hwcomposer/Android.mk hardware/libhardware/modules/hwcomposer/Android.mk
--- hardware/libhardware/modules/hwcomposer/Android.mk	2014-03-01 07:48:44.295130354 +0000
+++ hardware/libhardware/modules/hwcomposer/Android.mk	2014-03-01 07:46:32.711754802 +0000
@@ -18,7 +18,7 @@
 # HAL module implemenation stored in
 # hw/<OVERLAY_HARDWARE_MODULE_ID>.<ro.product.board>.so
 include $(CLEAR_VARS)
-
+LOCAL_CLANG := true
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 LOCAL_SHARED_LIBRARIES := liblog libEGL
 LOCAL_SRC_FILES := hwcomposer.cpp
diff -ruN hardware/libhardware/modules/local_time/Android.mk hardware/libhardware/modules/local_time/Android.mk
--- hardware/libhardware/modules/local_time/Android.mk	2014-03-01 07:48:44.295130354 +0000
+++ hardware/libhardware/modules/local_time/Android.mk	2014-03-01 07:46:32.711754802 +0000
@@ -24,7 +24,7 @@
 #
 # The format of the name is local_time.<hardware>.so
 include $(CLEAR_VARS)
-
+LOCAL_CLANG := true
 LOCAL_MODULE := local_time.default
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 LOCAL_SRC_FILES := local_time_hw.c
diff -ruN hardware/libhardware/modules/nfc/Android.mk hardware/libhardware/modules/nfc/Android.mk
--- hardware/libhardware/modules/nfc/Android.mk	2014-03-01 07:48:44.295130354 +0000
+++ hardware/libhardware/modules/nfc/Android.mk	2014-03-01 07:46:32.711754802 +0000
@@ -15,7 +15,7 @@
 LOCAL_PATH := $(call my-dir)
 
 include $(CLEAR_VARS)
-
+LOCAL_CLANG := true
 LOCAL_MODULE := nfc.default
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 LOCAL_SRC_FILES := nfc_pn544_example.c
diff -ruN hardware/libhardware/modules/nfc-nci/Android.mk hardware/libhardware/modules/nfc-nci/Android.mk
--- hardware/libhardware/modules/nfc-nci/Android.mk	2014-03-01 07:48:44.295130354 +0000
+++ hardware/libhardware/modules/nfc-nci/Android.mk	2014-03-01 07:46:32.711754802 +0000
@@ -15,7 +15,7 @@
 LOCAL_PATH := $(call my-dir)
 
 include $(CLEAR_VARS)
-
+LOCAL_CLANG := true
 LOCAL_MODULE := nfc_nci.default
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 LOCAL_SRC_FILES := nfc_nci_example.c
diff -ruN hardware/libhardware/modules/power/Android.mk hardware/libhardware/modules/power/Android.mk
--- hardware/libhardware/modules/power/Android.mk	2014-03-01 07:48:44.295130354 +0000
+++ hardware/libhardware/modules/power/Android.mk	2014-03-01 07:46:32.711754802 +0000
@@ -15,7 +15,7 @@
 LOCAL_PATH := $(call my-dir)
 
 include $(CLEAR_VARS)
-
+LOCAL_CLANG := true
 LOCAL_MODULE := power.default
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 LOCAL_SRC_FILES := power.c
diff -ruN hardware/libhardware/modules/usbaudio/Android.mk hardware/libhardware/modules/usbaudio/Android.mk
--- hardware/libhardware/modules/usbaudio/Android.mk	2014-03-01 07:48:44.295130354 +0000
+++ hardware/libhardware/modules/usbaudio/Android.mk	2014-03-01 07:46:32.711754802 +0000
@@ -15,7 +15,7 @@
 LOCAL_PATH := $(call my-dir)
 
 include $(CLEAR_VARS)
-
+LOCAL_CLANG := true
 LOCAL_MODULE := audio.usb.default
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 LOCAL_SRC_FILES := \
diff -ruN hardware/libhardware/tests/camera2/Android.mk hardware/libhardware/tests/camera2/Android.mk
--- hardware/libhardware/tests/camera2/Android.mk	2014-03-01 07:48:44.295130354 +0000
+++ hardware/libhardware/tests/camera2/Android.mk	2014-03-01 07:46:32.707754821 +0000
@@ -1,6 +1,6 @@
 LOCAL_PATH:= $(call my-dir)
 include $(CLEAR_VARS)
-
+LOCAL_CLANG := true
 LOCAL_SRC_FILES:= \
 	camera2.cpp \
 	camera2_utils.cpp \
diff -ruN hardware/libhardware/tests/hwc/Android.mk hardware/libhardware/tests/hwc/Android.mk
--- hardware/libhardware/tests/hwc/Android.mk	2014-03-01 07:48:44.299130335 +0000
+++ hardware/libhardware/tests/hwc/Android.mk	2014-03-01 07:46:32.707754821 +0000
@@ -1,11 +1,13 @@
 LOCAL_PATH:= $(call my-dir)
 
 include $(CLEAR_VARS)
+LOCAL_CLANG := true
 LOCAL_MODULE := libcnativewindow 
 LOCAL_SRC_FILES := cnativewindow.c util.c
 include $(BUILD_STATIC_LIBRARY)
 
 include $(CLEAR_VARS)
+LOCAL_CLANG := true
 LOCAL_MODULE := hwc-test-arrows
 LOCAL_SRC_FILES := test-arrows.c
 LOCAL_STATIC_LIBRARIES := libcnativewindow
diff -ruN hardware/libhardware/tests/keymaster/Android.mk hardware/libhardware/tests/keymaster/Android.mk
--- hardware/libhardware/tests/keymaster/Android.mk	2014-03-01 07:48:44.299130335 +0000
+++ hardware/libhardware/tests/keymaster/Android.mk	2014-03-01 07:46:32.707754821 +0000
@@ -2,7 +2,7 @@
 
 LOCAL_PATH:= $(call my-dir)
 include $(CLEAR_VARS)
-
+LOCAL_CLANG := true
 LOCAL_SRC_FILES:= \
     keymaster_test.cpp
 
diff -ruN hardware/libhardware/tests/nusensors/Android.mk hardware/libhardware/tests/nusensors/Android.mk
--- hardware/libhardware/tests/nusensors/Android.mk	2014-03-01 07:48:44.299130335 +0000
+++ hardware/libhardware/tests/nusensors/Android.mk	2014-03-01 07:46:32.707754821 +0000
@@ -1,6 +1,6 @@
 LOCAL_PATH:= $(call my-dir)
 include $(CLEAR_VARS)
-
+LOCAL_CLANG := true
 LOCAL_SRC_FILES:= \
 	nusensors.cpp
 diff -ruN hardware/libhardware_legacy/Android.mk hardware/libhardware_legacy/Android.mk
--- hardware/libhardware_legacy/Android.mk	2014-03-01 07:59:42.695882468 +0000
+++ hardware/libhardware_legacy/Android.mk	2014-03-01 07:59:28.391955080 +0000
@@ -5,7 +5,7 @@
 
 SAVE_MAKEFILES := $(call all-named-subdir-makefiles,$(legacy_modules))
 LEGACY_AUDIO_MAKEFILES := $(call all-named-subdir-makefiles,audio)
-
+LOCAL_CLANG := true
 LOCAL_PATH:= $(call my-dir)
 include $(CLEAR_VARS)
 
@@ -26,7 +26,7 @@
 
 # static library for librpc
 include $(CLEAR_VARS)
-
+LOCAL_CLANG := true
 LOCAL_MODULE:= libpower
 
 LOCAL_SRC_FILES += power/power.c
@@ -35,7 +35,7 @@
 
 # shared library for various HALs
 include $(CLEAR_VARS)
-
+LOCAL_CLANG := true
 LOCAL_MODULE := libpower
 
 LOCAL_SRC_FILES := power/power.c
diff -ruN hardware/libhardware_legacy/audio/Android.mk hardware/libhardware_legacy/audio/Android.mk
--- hardware/libhardware_legacy/audio/Android.mk	2014-03-01 07:59:42.695882468 +0000
+++ hardware/libhardware_legacy/audio/Android.mk	2014-03-01 07:59:28.395955060 +0000
@@ -6,6 +6,7 @@
 LOCAL_PATH := $(call my-dir)
 include $(CLEAR_VARS)
 
+LOCAL_CLANG := true
 LOCAL_SRC_FILES := \
     AudioHardwareInterface.cpp \
     audio_hw_hal.cpp
@@ -18,6 +19,7 @@
 
 include $(CLEAR_VARS)
 
+LOCAL_CLANG := true
 LOCAL_SRC_FILES := \
     AudioPolicyManagerBase.cpp \
     AudioPolicyCompatClient.cpp \
@@ -37,6 +39,7 @@
 # policy code
 include $(CLEAR_VARS)
 
+LOCAL_CLANG := true
 LOCAL_SRC_FILES := \
     AudioPolicyManagerDefault.cpp
 

